collector/src/main/java/com/bitmovin/analytics/data/BackendFactory.java

```diff
@@ -3,10 +3,18 @@
 import android.content.Context;
 
 import com.bitmovin.analytics.BitmovinAnalyticsConfig;
+import com.bitmovin.analytics.retryBackend.RetryBackend;
+import android.os.Handler;
+
 
 public class BackendFactory {
 
     public Backend createBackend(BitmovinAnalyticsConfig config, Context context) {
-        return new HttpBackend(config.getConfig(), context);
+        HttpBackend httpBackend = new HttpBackend(config.getConfig(), context);
+        if(!config.getConfig().getTryResendDataOnFailedConnection() ){
+           return httpBackend;
+       }
+
+        return new RetryBackend(httpBackend, new Handler());
```

Comment:
could this be a problem as we don't have access to the `Handler` instance anymore? Is unregistering stuff from the handler enough to not cause memory leaks?

Reply:
It was moved out for easier testing... And new Handler instance is created in factory inside `retryBackend` constructor But that class has destructor so I assume it should be fine

Reply:
Since nobody has a reference to it the GC should collect it. So if only RetryBackend + Handler reference each other they end up outside the alive set so I don't think there can be a memory issue